src/module.c

```diff
@@ -8880,8 +8924,14 @@ void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key, int dbid)
         if ((sub->event_mask & type) &&
             (sub->active == 0 || (sub->module->options & VALKEYMODULE_OPTIONS_ALLOW_NESTED_KEYSPACE_NOTIFICATIONS))) {
             ValkeyModuleCtx ctx;
-            moduleCreateContext(&ctx, sub->module, VALKEYMODULE_CTX_TEMP_CLIENT);
-            selectDb(ctx.client, dbid);
+            if (c == NULL) {
```

Comment:
Rather than passing in `client*` to this module API, can you simply check if `server.current_client` is NULL or not?